modules/webbrowser/processors/webbrowserprocessor.cpp

```diff
@@ -166,24 +196,34 @@ WebBrowserProcessor::~WebBrowserProcessor() {
     // called
     browserClient_->SetRenderHandler(NULL);
 }
-    
+
 void WebBrowserProcessor::deserialize(Deserializer& d) {
     Processor::deserialize(d);
-    
+
     for (auto prop : *this) {
-        if (prop == &url_ || prop == &reload_ || prop == &addPropertyGroup_ ||
-            prop == &type_ || prop == &propertyHtmlId_ || prop == &add_) {
+        if (prop == &sourceType_ || prop == &fileName_ || prop == &url_ || prop == &reload_ ||
+            prop == &addPropertyGroup_ || prop == &type_ || prop == &propertyHtmlId_ ||
+            prop == &add_) {
             continue;
         }
         browserClient_->propertyCefSynchronizer_->startSynchronize(prop, prop->getIdentifier());
     }
     // Must reload page to connect property with Frame, see PropertyCefSynchronizer::OnLoadEnd
-    browser_->GetMainFrame()->LoadURL(url_.get());
+    if (sourceType_.get() == SourceType::LocalFile) {
```

Comment:
Getting the URL  could be moved to a separate function since it is repeated a couple of times

Reply:
Agree! Done!